### DISCUSSION TITLE
[BUG] Duplicate photos showing up on grid (mobile and web)

### DISCUSSION THREAD

**Discussion by AndresSalinasB** [immich-app/immich#6463] [Q&A] [answered]
### The bug

Every photo that I take with my mobile device now it's showing duplicated in the web and in the mobile app. 

In the mobile app one version of the photo has a cloud with symbol with a checkmark in it and the other one has the cloud symbol without the check.
![Screenshot 2024-01-17 at 9 24 54 AM](https://github.com/immich-app/immich/assets/2312471/1d5bb1ec-25d5-4b17-904e-12add9975dc2)
![Screenshot_20240117-092509](https://github.com/immich-app/immich/assets/2312471/1326de7c-7974-48e8-a024-f2cd410a702a)

Attaching screenshots.

I don't have a specific docker-compose/.env file since this is hosted as a TrueNAS app. What info should I provide in this case?

### The OS that Immich Server is running on

TrueNAS

### Version of Immich Server

v1.92.1

### Version of Immich Mobile App

v1.92.0

### Platform with the issue

- [X] Server
- [X] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
Using TrueNAS app.
```

### Your .env content

```Shell
Using TrueNAS app.
```

### Reproduction steps

```bash
1. Take a new picture
2. Wait for upload and after some time it shows up duplicated.
```

**Comment by aviv926** [comment]
Are you using the External Libraries option?
Is your timeline shared with a partner?
Have you tried clearing the browser cache and logging in through the browser interface again?

**Comment by AndresSalinasB** [comment]
Yes to both. I have an external library and also I'm sharing my photos with a partner.

**Comment by AndresSalinasB** [comment]
now that you mention it, the duplication starts as soon as the daily job to scan the libraries gets executed. it seems that they're getting uploaded in both the external and the default library and that's causing duplication?

is there a way to merge this two libraries together?

**Comment by AndresSalinasB** [comment]
to add more to it, it seems that all of this originated when I moved my installation from truecharts to truenas and I just imported the old library as an external library but with the same location as the default library. This may have caused the issue.

**Comment by jrasm91** [comment, marked as answer]
Right now, deduping only happens across libraries. If you have the same photo in two libraries, you will see two copies in your timeline. If the same photo is owned and shared via partner sharing, it will also appear twice on the timeline. This is something we hope to improve in the future.

**Comment by Gadgitmatic** [comment]
I made a similar mistake. I want to be able to upload or **sync my mobile photos to my existing external library**
![Screenshot 2024-02-07 204821](https://github.com/immich-app/immich/assets/28793787/fb16fec9-49e3-4fd0-9321-81e4f2f803b0)

Is there a recommended method for doing so? I am also using TrueNAS Scale. 

If anyone is using TrueNAS and can elaborate on how they are managing their Immich storage that would be super helpful!

**Reply by aisbergde** [reply]
I had the same issue, when I started to use immich. Finally, I just stopped to use the immich upload and switched to use only external libraries in combination with automated upload into external libraries.

The main reason for my decision was: if I delete assets from the default library not via GUI, but directly in the file system, then the are still shown in immich GUI. They are offline, but there is nothing implemented to remove offline assets from the default libraries. There are some python scripts, doing this, but this is a manual process.

I understand the idea, that the default library should be managed only by immich, but this concept has benefits and downsides.

**Reply by Gadgitmatic** [reply]
I can't login to my Immich instance to check at the moment, but isn't there a rescan feature to check for deletions/etc?

**Reply by aisbergde** [reply]
rescan for deleted assets currently happens only in external libraries. The concept of the default library is, that their assets are managed only by immich. There are workarounds to handle this, but it requires some scripts or multi step operations.
